parsers/options.js

```diff
@@ -0,0 +1,24 @@
+export default class OptionsParser {
+    parse(str) {
+        let trimmed = str.trim();
+        if (trimmed[0] === '/') trimmed = trimmed.substr(1);
+        if (trimmed[trimmed.length-1] === '/') trimmed = trimmed.substr(0, trimmed.length - 1);
+
+        /* if his segment is prefaced with a 'opt=', we know we're parsing options */
+        if ( trimmed.substr(0, 4) === 'opt=' ) {
```

Comment:
I wonder if we should use less characters here. 
You were using ! Before? 
!gh !d !g0
Are we likely to add other configuration options? 
Maybe this group of options are actually specifically grid options. Is it possible to mix and match?
We could reserve a character for options/settings and then use a specific character with each different option/setting type. Something like !g for grid options or !f for font options (if we ever had such an option)

So final command would be !gd !g0 !gh or mix and match with !gdh for dark and 50%
Thoughts?

Reply:
I think my logic for using `opt` is that these options are something that should be set as a group and only once per call, unlike the SVG symbols which we'd be using a bunch - so we could save the `!` for something we add later that's used more often within a url

Reply:
It is a deviation from what we've normally been doing, though, so I'd be fine with changing it back to !

Reply:
the @ symbol has been used for an option already (zoom), might want to combine?

Reply:
yeah could do that. maybe the @ symbol should be the symbol for options.
We could get this down to 2-3 characters

Options

* @1 zoom level 1
* @2 zoom level 2
* @3 zoom level 3
* @d dark mode
* @n no grid
* @h half opacity grid

Combining

* @2dh - darkmode, zoom level 2, half opacity grid

?

Reply:
I think the combination makes it a bit unnecessarily complicated, but I like the other options.
iirc the max limit we want to keep to is 2k characters in the url. Because these options are only being used once each per url, I think keeping them separate for now is probably fine

Reply:
like `/@2/@d/@h` ? Looks fine but means you have to switch your keyboard on your mobile phone back and forth. `/@2dh` is a lot easier on a phone. We do something similar with the tokens where you can mix and match size and color flags in any order as you please

Eg. `/C10Lr-goblin`

Reply:
That's fine then, if we're grouping other options together. Will make update

Reply:
could do both ways: do long options for PC, short for phone.